src/wasm/wasm-type.cpp

```diff
@@ -3036,14 +3105,22 @@ void canonicalizeEquirecursive(CanonicalizationState& state) {
 }
 
 void canonicalizeNominal(CanonicalizationState& state) {
+  // TODO: clear recursion groups once we are no longer piggybacking the
+  // isorecursive system on the nominal system.
+  // if (typeSystem != TypeSystem::Isorecursive) {
+  //   for (auto& info : state.newInfos) {
+  //     assert(info->recGroup == nullptr && "unexpected recursion group");
+  //   }
+  // }
+
   // Nominal types do not require separate canonicalization, so just validate
   // that their subtyping is correct.
 
 #if TIME_CANONICALIZATION
   auto start = std::chrono::steady_clock::now();
 #endif
 
-  assert(typeSystem == TypeSystem::Nominal);
+  // assert(typeSystem == TypeSystem::Nominal);
```

Comment:
Was this intentional? Or should it be removed?

Reply:
Yeah this can probably just be removed. It's not a very valuable assertion.